### PR TITLE
add a basic workflow

### LINKED ISSUE #579: add CI to tutorial repo
add checking that examples compile and run with success

### PR DESCRIPTION
**Pull request by haichangsi**
all checks here - https://github.com/haichangsi/distributed-ranges-tutorial/actions/runs/6624423249

### CODE REVIEW

File: scripts/build_run.sh [hunk 1]
```diff
@@ -4,6 +4,9 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
+export CXX=icpx
+export CC=icx
+
 cmake -B build -DENABLE_SYCL=on
```
Comment (mateuszpn) [inline scripts/build_run.sh]: ENABLE_SYCL has no effect in project's CMakeList.txt.
The examples require SYCL compiler to be available, not enabling SYCL makes no sense. Consider adding a user friendly message about missing SYCL compiler, if icpx is not available (in CMakeList.txt probably).

File: .github/workflows/docker_workflow.yml [hunk 2]
```diff
@@ -28,10 +41,6 @@ jobs:
         uses: actions/checkout@v3
       - name: Setup cmake
         uses: jwlawson/actions-setup-cmake@v1.13
-      - name: Setup Python
-        uses: actions/setup-python@v4
-        with:
-          python-version: '3.10'
       - name: Build examples
         run: |
           cmake -B build -DENABLE_SYCL=on
```
Comment (mateuszpn) [inline .github/workflows/docker_workflow.yml]: ENABLE_SYCL has no effect in project's CMakeList.txt.

**Review by mateuszpn — CHANGES_REQUESTED** [review]
Thanks for fixing all the typos.
Some comments below

File: scripts/build_run.sh (hunk 1 above)
Comment (haichangsi) [reply scripts/build_run.sh]: The script was left only to make it easier to run all with one command, I should probably make another pr with it, let's maybe leave it as it is (with no flag - thx) and I will enhance it's functionality in another PR ;)

File: .github/workflows/docker_workflow.yml (hunk 2 above)
Comment (haichangsi) [reply .github/workflows/docker_workflow.yml]: right, thx, Done

**Pull request merged**